src/Stratis.Bitcoin.Features.PoA/FederationManager.cs

```diff
@@ -69,92 +92,102 @@ public void Initialize()
+            this.IsFederationMember = this.federationMembers.Any(x => x.PubKey == this.CurrentFederationKey?.PubKey);
         }
 
-        /// <summary>Provides up to date list of federation members.</summary>
-        /// <remarks>
-        /// Blocks that are not signed with private keys that correspond
-        /// to public keys from this list are considered to be invalid.
-        /// </remarks>
-        public List<PubKey> GetFederationMembers()
+        /// <inheritdoc />
+        public List<IFederationMember> GetFederationMembers()
         {
             lock (this.locker)
             {
-                return new List<PubKey>(this.federationMembers);
+                return new List<IFederationMember>(this.federationMembers);
             }
         }
 
-        public void AddFederationMember(PubKey pubKey)
+        public virtual void AddFederationMember(IFederationMember federationMember)
         {
             lock (this.locker)
             {
-                if (this.federationMembers.Contains(pubKey))
+                if (this.federationMembers.Contains(federationMember))
                 {
                     this.logger.LogTrace("(-)[ALREADY_EXISTS]");
                     return;
                 }
 
-                this.federationMembers.Add(pubKey);
+                this.federationMembers.Add(federationMember);
 
-                this.SaveFederationKeys(this.federationMembers);
+                this.SaveFederation(this.federationMembers);
                 this.SetIsFederationMember();
 
-                this.logger.LogInformation("Federation member '{0}' was added!", pubKey.ToHex());
+                this.logger.LogInformation("Federation member '{0}' was added!", federationMember);
             }
 
-            this.signals.Publish(new FedMemberAdded(pubKey));
+            this.signals.Publish(new FedMemberAdded(federationMember));
         }
 
-        public void RemoveFederationMember(PubKey pubKey)
+        public void RemoveFederationMember(IFederationMember federationMember)
         {
             lock (this.locker)
             {
-                this.federationMembers.Remove(pubKey);
+                this.federationMembers.Remove(federationMember);
 
-                this.SaveFederationKeys(this.federationMembers);
+                this.SaveFederation(this.federationMembers);
                 this.SetIsFederationMember();
 
-                this.logger.LogInformation("Federation member '{0}' was removed!", pubKey.ToHex());
+                this.logger.LogInformation("Federation member '{0}' was removed!", federationMember);
             }
 
-            this.signals.Publish(new FedMemberKicked(pubKey));
+            this.signals.Publish(new FedMemberKicked(federationMember));
         }
 
-        private void SaveFederationKeys(List<PubKey> pubKeys)
+        protected virtual void SaveFederation(List<IFederationMember> federation)
```

Comment:
Why are these `virtual`?

Reply:
There is a `CollateralFederationManager` that overrides them